tests/lax_numpy_ufuncs_test.py

```diff
@@ -147,6 +155,22 @@ def test_reduce(self, func, nin, nout, identity, shape, axis, dtype):
     self._CheckAgainstNumpy(jnp_fun, np_fun, args_maker)
     self._CompileAndCheck(jnp_fun, args_maker, check_cache_misses=False)  # TODO(jakevdp): why the cache misses?
 
+  @jtu.sample_product(
+      FASTPATH_FUNCS,
+      [{'shape': shape, 'axis': axis}
+       for shape in nonscalar_shapes
+       for axis in range(-len(shape), len(shape))],
+      dtype=jtu.dtypes.floating,
+  )
+  def test_reduce_fastpath(self, func, nin, nout, identity, shape, axis, dtype, reducer, accumulator):
+    del accumulator  # unused
+    if (nin, nout) != (2, 1):
+      self.skipTest(f"reduce requires (nin, nout)=(2, 1); got {(nin, nout)=}")
+    rng = jtu.rand_default(self.rng())
+    args = (rng(shape, dtype),)
+    jnp_fun = partial(jnp.frompyfunc(func, nin=nin, nout=nout, identity=identity).reduce, axis=axis)
+    self.assertEqual(get_single_primitive(jnp_fun, *args), reducer)
```

Comment:
Maybe we should also check that we _don't_ hit the fastpath for some well-chosen `func`s, i.e. ones that would _almost_ hit the fastpath.

Reply:
Done